jsonrpc/eth_endpoint_test.go

```diff
@@ -214,104 +281,78 @@ func TestEth_Block_BlockNumber(t *testing.T) {
 				Topics:    topics,
-			}, true},
-			{"Reference blocks not found", &LogFilter{
-				fromBlock: 10,
-				toBlock:   20,
+			},
+			false, 3},
+		{"Found matching logs, fromBlock == toBlock",
+			&LogFilter{
+				fromBlock: 2,
+				toBlock:   2,
 				Topics:    topics,
-			}, true},
-		}
-
-		// Setup //
-		store := newMockBlockStore()
-
-		dispatcher := newTestDispatcher(hclog.NewNullLogger(), store)
-
-		store.getHeaderByNumberCallback = func(blockNumber uint64) (*types.Header, bool) {
-			return &types.Header{
-				Number:       blockNumber,
-				ReceiptsRoot: types.StringToHash(strconv.FormatUint(blockNumber, 10)),
-			}, true
-		}
-
-		store.getReceiptsByHashCallback = func(hash types.Hash) ([]*types.Receipt, error) {
-
-			switch hash {
-			case types.StringToHash(strconv.FormatUint(1, 10)):
-				return store.generateReceipts(
-					generateReceiptsParams{
-						numReceipts: 1,
-						numTopics:   1,
-						numLogs:     2,
-					},
-				), nil
-			case types.StringToHash(strconv.FormatUint(2, 10)):
-				return store.generateReceipts(
-					generateReceiptsParams{
-						numReceipts: 2,
-						numTopics:   5,
-						numLogs:     1,
-					},
-				), nil
-			case types.StringToHash(strconv.FormatUint(3, 10)):
-				return store.generateReceipts(
-					generateReceiptsParams{
-						numReceipts: 3,
-						numTopics:   10,
-						numLogs:     5,
-					},
-				), nil
-			}
-
-			return nil, nil
-		}
+			},
+			false, 1},
+		{"No logs found", &LogFilter{
+			fromBlock: 4,
+			toBlock:   5,
+			Topics:    topics,
+		}, false, 0},
+		{"Invalid block range", &LogFilter{
+			fromBlock: 10,
+			toBlock:   5,
+			Topics:    topics,
+		}, true, 0},
+	}
 
-		for index, testCase := range testTable {
+	// setup test
+	store := &mockBlockStore2{}
+	store.topics = []types.Hash{topic1, topic2, topic3}
+	for i := 0; i < 4; i++ {
+		store.add(&types.Block{
+			Header: &types.Header{
+				Number: uint64(i),
+				Hash:   types.StringToHash(strconv.Itoa(i)),
+			},
+		})
+	}
+	dispatcher := newTestDispatcher(hclog.NewNullLogger(), store)
 
-			if index == 2 {
-				store.getHeaderByNumberCallback = func(blockNumber uint64) (*types.Header, bool) {
-					return nil, false
+	for _, testCase := range testTable {
+		t.Run(testCase.name, func(t *testing.T) {
+			foundLogs, logError := dispatcher.endpoints.Eth.GetLogs(
+				testCase.filterOptions,
+			)
+
+			if logError != nil && !testCase.shouldFail {
+				// If there is an error and test isn't expected to fail
+				t.Fatalf("Error: %v", logError)
+			} else if !testCase.shouldFail {
+				switch testCase.name {
+				case "Found matching logs, fromBlock < toBlock":
+					assert.Lenf(t, foundLogs, testCase.expectedLength, "Invalid number of logs found")
+				case "Found matching logs, fromBlock == toBlock":
+					assert.Lenf(t, foundLogs, testCase.expectedLength, "Invalid number of logs found")
+				case "No logs found":
+					assert.Lenf(t, foundLogs, testCase.expectedLength, "Invalid number of logs found")
```

Comment:
Same args are passed in every cases, can we remove switch? That was intended to do that.